### PR TITLE
Adds name attribute to <testcase> elements in junit output (#575)

### PR DESCRIPTION
**Pull request by akbertram**
Improved output: 
http://build.renjin.org/job/Replacement-Packages/job/testthat/11/testReport/

### CODE REVIEW

**Comment by akbertram** [comment]
Fixes Issue #575

**Comment by akbertram** [comment]
@hadley I believe I've fixed the test failures but it looks like neither AppVeyor nor Travis are rebuilding.

File: R/reporter-junit.R [hunk 1]
```diff
@@ -89,7 +89,8 @@ JunitReporter <- R6::R6Class("JunitReporter", inherit = Reporter,
       name <- test %||% "(unnamed)"
       testcase <- xml2::xml_add_child(self$suite, "testcase",
        time = toString(time),
-       classname = paste0(classnameOK(context), '.', classnameOK(name))
```
Comment (hadley) [inline R/reporter-junit.R]: Would you mind fixing the indenting while you're in here?

**Comment by akbertram** [comment]
@hadley When I run `test_package("testthat")` all tests pass. But if I run `devtools::check()` . This seems to be because the junit reporter includes stack depth in the error message, which can vary depending on how the test is invoked:

```.xml
 <failure type="failure" message="expectation_failure in structure(list(fail()), start_frame = 19): Failure has been forced&#10;&#10; @reporters/tests.R#8"/>
```

I would suggest removing the `start_frame` attribute from the output anyway, it makes the deparsed function call harder to read?

**Comment by hadley** [comment]
Yeah, that looks like a mistake anyway - "structure(list(fail()), start_frame = 19)" looks like a deparsed object, not something deliberately included. Can you remove it?

**Comment by hadley** [comment]
I can't really follow the logic of the code that currently generates the message, and it's not clear to me what it should be, so I trust your judgement as to what would be more helpful.

**Comment by akbertram** [comment]
Phew. That seems to work now. I've added one more commit that allows CI systems to override the reporter used for test_check() and specify and output location for the junit reporter.

**Comment by hadley** [comment]
Looks good. Now just needs a bullet in NEWS.md, and I can merge.

**Comment by hadley** [comment]
Thanks!

**Comment by antonytanj** [comment]
What is the process to publish a new version of this package with these changes?

**Pull request merged**